### PR TITLE
HPCC-20903 Send html page for session reload when invalid session

### PR DESCRIPTION
**Pull request by wangkx**
When ECLWatch page is closed without logging out, the ESP session
may be timed out in ESP, but, not in Internet browser. If the user
reopens ECLWatch by typing in the IP, the existing ESP returns a
JSON error. This fix replaces the JSON error with a html page.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
@RussWhitehead @miguelvazq please review.

**Review by RussWhitehead — COMMENTED** [review]
Looks good, a few minor comments

File: esp/bindings/http/platform/httpservice.cpp [hunk 1]
```diff
@@ -1530,6 +1530,123 @@ void CEspHttpServer::resetSessionTimeout(EspAuthRequest& authReq, unsigned sessi
...
+
+                ".visible {"
+                    "display: none;"
+                "}"
+            "</style>"
+            "</head>"
+            "<body>"
+                "<div id=\"container\" class=\"container\">"
+                    "<div class=\"formContainer\">");
+        content.appendf("<img id=\"logo\" src=\"%s%s\" />", espURL.str(), authReq.authBinding->queryLoginLogoURL());
```
[the remark this answers is not in the document]
Comment (wangkx) [reply esp/bindings/http/platform/httpservice.cpp]: The indentation here is a better read the HTML code.

File: initfiles/componentfiles/configxml/esp.xsd.in [hunk 2]
```diff
@@ -412,6 +412,13 @@
                                 </xs:appinfo>
                             </xs:annotation>
                         </xs:attribute>
+                        <xs:attribute name="loginLogoURL" type="xs:string" use="optional" default="/esp/files/eclwatch/img/Loginlogo.png">
```
[the remark this answers is not in the document]
Comment (wangkx) [reply initfiles/componentfiles/configxml/esp.xsd.in]: Someone may want to change this if the img files are installed into another location. We do the same as for logonURL, etc.

File: esp/bindings/http/platform/httpbinding.cpp [hunk 3]
```diff
@@ -389,6 +394,11 @@ void EspHttpBinding::readAuthDomainCfg(IPropertyTree* procCfg)
             loginURL.set(_loginURL);
         else
             loginURL.set(DEFAULT_LOGIN_URL);
+        const char* _loginLogoURL = authDomainTree->queryProp("@loginLogoURL");
```
[the remark this answers is not in the document]
Comment (wangkx) [reply esp/bindings/http/platform/httpbinding.cpp]: This is in readAuthDomainCfg() which will not be called if the previous _loginLogoURL is set ( line 312).

**Comment by RussWhitehead** [comment]
@wangkx Looks good to me
@miguelvazq Waiting for your review

**Comment by wangkx** [comment]
@miguelvazq please double check the html code.

File: esp/bindings/http/platform/httpservice.cpp [hunk 4]
```diff
@@ -1530,6 +1530,123 @@ void CEspHttpServer::resetSessionTimeout(EspAuthRequest& authReq, unsigned sessi
...
+                "img {"
+                    "display: block;"
+                    "margin: auto;"
+                "}"
+
+                "p {"
+                    "text-align: center"
+                "}"
+
+                "#button {"
```
Comment (miguelvazq) [inline esp/bindings/http/platform/httpservice.cpp]: Remove button style

File: esp/bindings/http/platform/httpservice.cpp [hunk 5]
```diff
@@ -1530,6 +1530,123 @@ void CEspHttpServer::resetSessionTimeout(EspAuthRequest& authReq, unsigned sessi
...
+                    "font-size: 16px;"
+                    "border-radius: 4px;"
+                    "cursor: pointer;"
+                    "-webkit-transition-duration: 0.4s;"
+                    "transition-duration: 0.4s;"
+                    "margin: auto;"
+                    "display: block;"
+                "}"
+
+               ".button:hover {"
```
Comment (miguelvazq) [inline esp/bindings/http/platform/httpservice.cpp]: Remove button style

File: esp/bindings/http/platform/httpservice.cpp [hunk 6]
```diff
@@ -1530,6 +1530,123 @@ void CEspHttpServer::resetSessionTimeout(EspAuthRequest& authReq, unsigned sessi
...
+                    "background-color: #13b1f9;"
+                    "color: white;"
+                "}"
+
+                ".visible {"
```
Comment (miguelvazq) [inline esp/bindings/http/platform/httpservice.cpp]: Remove .visible class

**Review by miguelvazq — CHANGES_REQUESTED** [review]

**Comment by wangkx** [comment]
@miguelvazq I just removed the code based on your comments. Please review.

**Comment by miguelvazq** [comment]
Looks good.

**Review by miguelvazq — APPROVED** [review]

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Pull request merged**